#!/usr/bin/env python
PACKAGE = "task_switch"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# gen.add("int_param", int_t, 0, "An Integer parameter", 50, 0, 100)
gen.add("activate_cbf", bool_t, 0, "A Boolean parameter", True)
gen.add("activate_fieldcbf", bool_t, 0, "A Boolean parameter", False)
gen.add("activate_chargecbf", bool_t, 0, "A Boolean parameter", False)
gen.add("activate_pcccbf", bool_t, 0, "A Boolean parameter", True)
gen.add("activate_staycbf", bool_t, 0, "A Boolean parameter", False)
gen.add("activate_collisioncbf", bool_t, 0, "A Boolean parameter", False)
gen.add("activate_umax", bool_t, 0, "A Boolean parameter", True)
gen.add("fieldcbf_slack_weight", double_t, 0, "A double parameter", 10000, 0, 10000)
gen.add("chargecbf_slack_weight", double_t, 0, "A double parameter", 10000., 0, 10000)
gen.add("pcccbf_slack_weight", double_t, 0, "A double parameter", 100000., 0, 100000)
gen.add("staycbf_slack_weight", double_t, 0, "A double parameter", 400., 0, 10000)
gen.add("gamma", double_t, 0, "A double parameter", 0.2, 0, 10)
gen.add("pcc_CBF_h_gain_k", double_t, 0, "A double parameter", 0.3, 0, 10)
gen.add("umax", double_t, 0, "A double parameter", 0.5, 0, 2.0)
# gen.add("str_param", str_t, 0, "A string parameter", "Hello World")
# gen.add("bool_param", bool_t, 0, "A Boolean parameter", True)


exit(gen.generate(PACKAGE, "task_switch", "cbf_params"))
